roofit/roofitcore/src/RooProdPdf.cxx

```diff
@@ -645,12 +640,11 @@ void RooProdPdf::factorizeProduct(const RooArgSet& normSet, const RooArgSet& int
 
     // Check if this PDF has dependents overlapping with one of the existing terms
     Bool_t done(kFALSE);
+    int j = 0;
     for (RooFIter lIter = termList.fwdIterator(),
-	ldIter = normList.fwdIterator(),
-	laIter = depAllList.fwdIterator();
+	ldIter = normList.fwdIterator();
       (termNormDeps = (RooArgSet*) ldIter.next(),
-       termAllDeps = (RooArgSet*) laIter.next(),
-       term = (RooArgSet*) lIter.next()); ) {
+       term = (RooArgSet*) lIter.next()); ++j) {
```

Comment:
And no possibly wrong casts. Nobody even checks that this cast is possible. Use static_cast to at least have some clue if it's at all possible to cast between those types.